sdk/eventhub/azure-eventhub/azure/eventhub/aio/_producer_async.py

```diff
@@ -134,53 +131,93 @@ async def _send_event_data(
         timeout_time: Optional[float] = None,
         last_exception: Optional[Exception] = None,
     ) -> None:
-        # TODO: Correct uAMQP type hints
         if self._unsent_events:
-            await self._open()
-            timeout = timeout_time - time.time() if timeout_time else 0
-            await self._handler.send_message_async(self._unsent_events[0], timeout=timeout)
-            self._unsent_events = None
+            await self._amqp_transport.send_messages_async(
+                self, timeout_time, last_exception, _LOGGER
```

Comment:
unless I missed it along the way, the timeout no longer needs default of 0 or delta between current time ?

Reply:
🤔 hmm, I must've missed this. Why do we no longer need this timeout?